lib/modeler/Linting.js

```diff
@@ -24,9 +24,7 @@ export default class Linting {
 
     const element = this._elementRegistry.get(id);
 
-    if (element !== this._canvas.getRootElement() && needsScrollToElement(this._canvas, element)) {
-      this._canvas.scrollToElement(element);
-    }
+    this._canvas.scrollToElement(element);
```

Comment:
If scroll is needed (including padding) is already covered in the `scrollToElement` implementation

Reply:
Are you sure this is the case? As far as I understand, `Canvas#scroll` is always called, and therefore the `canvas.viewbox.changing` event is always fired, which means every time an error is clicked, the annotations disappear for a bit. This was the original issue I was trying to solve.

Reply:
This is no longer a problem. I think we changed the overlay behavior recently so they are never hidden
![Recording 2023-08-22 at 09 30 36](https://github.com/camunda/linting/assets/21984219/ad250b77-3d86-4287-9a92-5fbd4f850ea5)

Reply:
Oh, I wasn't aware of this!

Reply:
I was not aware of this, too. When and how did we change it?

Reply:
Asking because I assumed this is general overlay behavior to hide as the canvas moves.

Reply:
I thought this was intentionally covered as "always show overlays" in the changelog: https://github.com/camunda/linting/blob/main/CHANGELOG.md#320

Reply:
But looking at the PR, this should only stop them from being hidden when we zoom out to much... Guess I assumed this bug was a feature

Reply:
never mind, the overlays are only shown because of our test infrastructure:

https://github.com/bpmn-io/bpmn-js/blob/9fb2daed42b8d7f280e9f7ad1300e31aed5a953d/test/helper/index.js#L97


Still, I think it is safe to always show overlays, I'll prepare a PR in diagram-js: https://github.com/bpmn-io/diagram-js/issues/798